foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_8_0/OrphedomosToFabric8Migration.java

```diff
@@ -0,0 +1,277 @@
+    }
+    private boolean isTestDockerPom(Model model){
+        return model.getParent().getArtifactId().endsWith("-tests") && model.getArtifactId().endsWith("-tests-docker");
+    }
+
+    private Xpp3Dom getConfiguration(Plugin plugin) {
+        Object pluginConfiguration = plugin.getConfiguration();
+        return (Xpp3Dom) pluginConfiguration;
+    }
+
+    private static boolean hasConfigurationItem(Xpp3Dom configuration, String configurationItem){
+        return configuration.getChildren(configurationItem).length > 0;
+    }
+
+    /**
+     * Function to validate whether migration should be performed or not.
+     * @param file file to validate
+     * @return true if migration should be performed
+     */
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        boolean shouldExecute = false;
+        boolean migrateConfiguration = false;
+        boolean migratePackaging = false;
+
+        Model model = getLocationAnnotatedModel(file);
+        if (isRootProjectPom(model) || isTestDockerPom(model)){
+            shouldExecute = false;
+        } else {
+            Plugin orphedomosPlugin = getPlugin(model, OLD_ARTIFACT_ID);
+            if (orphedomosPlugin != null){
+                migrateConfiguration = true;
+            }
+            if(containsOrphedomosPackaging(file)){
+                migratePackaging = true;
+            }
+            shouldExecute = migrateConfiguration || migratePackaging;
+        }
+
+        return shouldExecute;
+    }
+
+    /**
+     * Performs the migration if the shouldExecuteOnFile() returns true
+     * @param file file to migrate
+     * @return isSuccessful - whether file was migrated successfully or not
+     */
+    @Override
+    protected boolean performMigration(File file) {
+        Model model = getLocationAnnotatedModel(file);
+        Plugin plugin  = getPlugin(model, OLD_ARTIFACT_ID);
+        boolean replacedConfig = false;
+        boolean replacedPackaging = false;
+        boolean isSuccessful = false;
+
+        Xpp3Dom config;
+
+        if (plugin != null){
+            config = getConfiguration(plugin);
+        } else {
+            config = null;
+            logger.info("Orphedomos plugin configuration not found. Skipping configuration migration.");
+        }
+
+        // check if pom file has orphedomos-maven-plugin configuration
+        if (config != null){
+            // grab the old imageVersion and imageName
+            IMAGE_VERSION = getImageVersion(config);
+            IMAGE_NAME = getImageName(config);
+
+            // Replace groupID and artifactID
+            PomModifications modifyPlugin = new PomModifications();
+            InputLocation startGroupId = plugin.getLocation("groupId");
+            InputLocation endGroupId = PomHelper.incrementColumn(startGroupId, OLD_GROUP_ID.length());
+
+            InputLocation startArtifactId = plugin.getLocation("artifactId");
```

Comment:
A: same here, these can be final strings as well

Reply:
done! added
```
    private static final String GROUP_ID = "groupId";
    private static final String ARTIFACT_ID = "artifactId";
```

Reply:
also added the following for consistency

```
    private static final String VERSION = "version";
    private static final String CONFIGURATION = "configuration";
    private static final String IMAGE_NAME = "imageName";
    private static final String IMAGE_VERSION = "imageVersion";
```